networkx/algorithms/centrality/decay.py

```diff
@@ -0,0 +1,115 @@
+import networkx as nx
+
+__all__ = ["decay_centrality"]
+
+
+def decay_centrality(G, u=None, delta=0.5, mode="all", weight=None):
+    r"""
+    Compute the decay centrality for nodes.
+    Decay centrality [1]_ of a node `u` is the decay parameter raised to power
+    sum of shortest distance between `u`and `n-1` reachable nodes. This
+    particular algorithm allows to manipulate the emphasis of reachable nodes
+    with respect to distance between the nodes.
+
+    .. math::
+        DC(u) = {\sum_{v=1}^{n - 1}} \delta ^{d(v, u)},
+        0 < \delta < 1
+    where `d(v, u)` is the shortest-path distance between `v` and `u`,
+    `delta` is the decay parameter with value between 0 and 1,
+    and `n` is the number of nodes that can reach `u`.
+
+    Notice that when `delta` approaches 0, the decay centrality approaches
+    degree and decay centrality becomes propostional to the degree centrality.
+
+    When `delta` approaches 1, the decay centrality approaches size of
+    component i.e the number of reachable nodes.
+
+    Parameters
+    ----------
+    G : graph
+      A NetworkX graph
+    u : node or list of nodes, optional
+      Returns values for only nodes in u.
+    delta : decay parameter, optional (default=0.5)
+      Use the specified decay parameter. It must be between 0 and 1.
+    mode : string, optional (default='all')
+      Ignored for undirected graphs.
+      Compute the decay centrality for nodes as if graph is undirected if mode
+      is 'all', or the decay centrality for incoming edges path if mode is 'in'
+      or the decay centrality for outgoing edges path if mode is 'out'.
+    weight : string or function, optional (default = None)
+      If None, every edge has weight/distance/cost 1.If a string, use this edge
+      attribute as the edge weight.Any edge attribute not present defaults
+      to 1. If this is a function, the weight of an edge is the value returned
+      by the function. The function must accept exactly three positional
+      arguments:
+      1. the two endpoints of an edge
+      2. the dictionary of edge attributes for that edge.
+      The function must return a number.
+    Returns
```

Comment:
The doc formatter wants a blank line before each "section" (underlined..  like Returns, Examples, etc).
We use 4 space indentation.
The "weight" parameter should use the same doc_string description as for `shortest_path_length`.
Each reference should have the two dots in front of it. 

You can check how all these documentation changes look in the generated html docs (one of the "tests" near the bottom of the list is called "documentation artifact" and if you click on the details it shows the html docs.